pkg/prometheus/operator.go

```diff
@@ -969,18 +968,22 @@ func (c *Operator) sync(key string) error {
 }
 
 func createSSetInputChecksum(p monitoringv1.Prometheus, c Config, ruleConfigMapNames []string) (string, error) {
-	json, err := json.Marshal(
-		struct {
-			P monitoringv1.Prometheus
-			C Config
-			R []string
-		}{p, c, ruleConfigMapNames},
+	hash, err := hashstructure.Hash(struct {
+		P monitoringv1.Prometheus
+		C Config
+		R []string `hash:"set"`
+	}{p, c, ruleConfigMapNames},
+		nil,
 	)
 	if err != nil {
-		return "", errors.Wrap(err, "failed to marshal Prometheus CRD and config to json")
+		return "", errors.Wrap(
+			err,
+			"failed to calculate combined hash of Prometheus CRD, config and"+
```

Comment:
nit: i think gofmt likes to put whitespace between operators and operands

Reply:
Thought so as well, but running `[space]+` through `gofmt` results in `+`.

Reply:
maybe a shorter error message would suffice, i.e. `hashing of prometheus CRD, config, and configmap names failed`?

Reply:
Shortened :+1: